bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/items/ItemRegistryImpl.java

```diff
@@ -347,6 +402,83 @@ public Item remove(String itemName, boolean recursive) {
         }
     }
 
+    private Metadata serializeTags(MetadataKey key, Set<String> tags) {
+        if (tags == null || tags.isEmpty()) {
+            return null;
+        }
+        String tagString = String.join(TAG_SEPARATOR, tags);
+        Metadata metadata = new Metadata(key, tagString, null);
+        return metadata;
+    }
+
+    private SortedSet<String> readTags(String itemName) {
+        MetadataKey key = new MetadataKey(TAG_NAMESPACE, itemName);
+        SortedSet<String> tags = new TreeSet<>();
+        if (metadataRegistry != null) {
+            Metadata metadata = metadataRegistry.get(key);
+            if (metadata != null) {
+                tags.addAll(Arrays.asList(metadata.getValue().split("\\" + TAG_SEPARATOR)));
```

Comment:
As the string `"\\"` depends on the `TAG_SEPARATOR`, I would also externalize it as a constant.